### PR TITLE
[AT-5354] Make the fully pathed `id` derivable for `cloud_id`

### PR DESCRIPTION
**Pull request by jesse-dds**
https://ccpo.atlassian.net/browse/AT-5354

Co-authored By: @okaycj-atat @graham-dds 

Fixes the cloud id by deriving the management group id from a `name` rather than just using the splatted id from the polling response.

This highlights ongoing issues around standardizing our usage of names and ids. We should be working towards this:

> Only store UUIDs or simple string identifiers in a portfolio's CSP data dictionary, and use these identifiers to build the path for "fully pathed" resources

https://github.com/dod-ccpo/atst/blob/staging/atat/domain/csp/cloud/README.md#best-practices-for-integrating-with-the-azure-rest-api

### CODE REVIEW

File: atat/jobs.py [hunk 1]
```diff
@@ -93,7 +93,9 @@ def do_create_application(csp: CloudProviderInterface, application_id=None):
         )
 
         app_result = csp.create_application(payload)
-        application.cloud_id = app_result.id
+        application.cloud_id = (
+            f"/providers/Microsoft.Management/managementGroups/{app_result.name}"
+        )
```
Comment (dandds) [inline atat/jobs.py:98]: Doesn't this go against the "only store UUIDs or simple string identifiers" principle?

File: atat/domain/csp/cloud/models.py [hunk 2]
```diff
@@ -372,6 +372,7 @@ def enforce_display_name_length(cls, name):
 
 class ManagementGroupCSPResponse(AliasModel):
     id: str
+    name: str
```
Comment (dandds) [inline atat/domain/csp/cloud/models.py:375]: Do we need and use both of these?

File: atat/jobs.py (hunk 1 above)
Comment (jesse-dds) [reply atat/jobs.py:98]: It still follows the principle as we no longer are storing the "id" received from the management group polling endpoint. We are instead storing the "name" and constructing the fully-pathed identifier from that. This issue is that in this context, the "id" received everywhere `create_management` group was used is a handle to an asynchronous operation, not an identifier for a management group.

File: atat/domain/csp/cloud/models.py (hunk 2 above)
Comment (jesse-dds) [reply atat/domain/csp/cloud/models.py:375]: Yes, there are several model classes that inherit from `ManagementGroupCSPResponse`. `name` was added to store the simpler "name" property from management group responses. But a lot of other code depends on the `id` property being there and removing it causes more cascading changes that are out of scope for this ticket.

**Comment by jesse-dds** [comment]
Maybe @graham-dds can explain better, this change was small but tricky and touches on issues with what we store from responses.

File: atat/jobs.py (hunk 1 above)
Comment (dandds) [reply atat/jobs.py:98]: This makes sense. I think we're talking about "storing" in different ways. I mean what we save to the database (which is the fully-pathed ID in this case) but I think you means what portion of the API response we keep as an identifier for the resource.

**Pull request merged**